### ISSUE TITLE
Questions About the GR00T N1.7 Model and Training Code

### ISSUE THREAD

**Issue by Peachypie98** [NVIDIA/Isaac-GR00T#701]
Hello,

I have several questions about the GR00T N1.7 model and the training code.

Q1. It seems that the default action horizon for the GR00T N1.7 model is 40. However, to train the model, we need to set up a modality configuration. If I want to use a 16-step action prediction horizon, the instructions say that I can set `delta_indices=list(range(0, 16))`.

In this case, does the model still output 40 action steps and then use only the first 16 steps by slicing? Or is the model’s action horizon actually changed from 40 to 16?

Q2. In `gr00t/experiment/launch_finetune.py`, I see the following code:
`config = get_default_config().load_dict(...)`

From what I understand, the configuration is divided into three main parts: model, data, and training.

I am confused about the image configuration in the model config versus the data config. In the model config, the image crop size is (230, 230) and the target size is (256, 256). In the data config, the image crop size is (244, 244) and the target size is (224, 224).

What is the final image input size passed to the model? I assume it is 256 × 256, but I would like to confirm.

Also, could you explain the image preprocessing pipeline? Is it something like this?
`original image→ data image crop size→ data image target size→ model image crop size→ model image target size`

Q3. I would like to save model checkpoints based on the best loss, instead of saving them every N training steps. I saw some code related to this, but it seems difficult to enable. Could you help me understand how to configure or implement checkpoint saving based on the best loss?

Q4. This is related to question 2. If I want to fix the image shape to 256 × 256 (no crop, no resize, use the dataset images as-is), how should I configure the image crop size and target size in both the model and data configurations?

Q5. I need to access the actual image tensor that is passed to the model because i would like to geed the same image into my custom model and use its output. Could you help me identify where in the codebase this image is prepared and passed into the model?

Thank You!
